src/patternfly/components/Switch/styles.scss

```diff
@@ -0,0 +1,107 @@
+@import "../../patternfly-utilities";
+
+.pf-c-switch {
+  --pf-c-switch--Height: calc(#{var(--pf-global--FontSize--md)} * #{var(--pf-global--LineHeight--md)});
+  --pf-c-switch--Width: calc(#{var(--pf-c-switch--Height)} * #{var(--pf-global--GoldenRatio)});
+  --pf-c-switch--LineHeight: var(--pf-global--LineHeight--md);
+  --pf-c-switch__toggle--Radius: var(--pf-c-switch--Height);
+  --pf-c-switch__toggle--Padding: calc(#{var(--pf-c-switch--Height)} / 2 * (1 - 1 / #{var(--pf-global--GoldenRatio)}));
+  --pf-c-switch__toggle--BackgroundColor: var(--pf-global--primary-color--100);
+  --pf-c-switch__toggle--m-off--BackgroundColor: var(--pf-global--BackgroundColor--dark-200);
+  --pf-c-switch__toggle--m-disabled--BackgroundColor: var(--pf-global--BackgroundColor--dark-transparent-200);
+  --pf-c-switch__thumb--Width: calc(#{var(--pf-c-switch--Height)} / #{var(--pf-global--GoldenRatio)});
+  --pf-c-switch__thumb--Height: var(--pf-c-switch__thumb--Width);
+  --pf-c-switch__thumb--Radius: var(--pf-c-switch__thumb--Width);
+  --pf-c-switch__thumb--BoxShadow: var(--pf-global--BoxShadow--md);
+  --pf-c-switch__thumb--BackgroundColor: var(--pf-global--BackgroundColor--light-100);
+  --pf-c-switch__thumb--m-disabled--BackgroundColor: var(--pf-global--BackgroundColor--light-300);
+  --pf-c-switch__thumb--RighPosition: calc(#{var(--pf-c-switch--Width)} - #{var(--pf-c-switch__thumb--Width)} - 2 * #{var(--pf-c-switch__toggle--Padding)});
+  --pf-c-switch__label--Color: var(--pf-global--Color--dark-100);
+  --pf-c-switch__label--m-disabled--Color: var(--pf-global--disabled-color--100);
+  --pf-c-switch__label--FontWeight: var(--pf-global--FontWeight--normal);
+  --pf-c-switch__label--FontSize: var(--pf-global--FontSize--md);
+  --pf-c-switch__label--LineHeight: var(--pf-global--LineHeight--md);
+  --pf-c-switch__label--PaddingLeft: var(--pf-global--spacer--md);
+  --pf-c-switch__label--PaddingRight: var(--pf-global--spacer--md);
+
+  position: relative;
+  display: inline-block;
+  height: var(--pf-c-switch--Height);
+  line-height: var(--pf-c-switch--LineHeight);
+  vertical-align: middle;
+
+  .pf-c-switch__input {
+    position: absolute;
+    width: 0;
```

Comment:
If we do decide to use a checkbox, setting `width: 0` prevents NVDA from seeing the checkbox. I think you can just remove this property, and it would work fine. It might be worth reviewing how other design systems handle properties for the checkbox, too.